### ISSUE TITLE
Configuration ignored with RepositoryStateMachineModelFactory

### ISSUE THREAD

**Issue by jkranes** [spring-projects/spring-statemachine#459]
When creating a state machine via @ EnableStateMachine and extending StateMachineConfigurerAdapter, any configuration set by means of overriding configure(StateMachineConfigurationConfigurer) is ignored when the state machine is constructed, if the state machine is obtained from a repository (e.g. MongoDB) via the RepositoryStateMachineModelFactory.

The reason for the problem appears to be the implementation code in AbstractStateMachineModelFactory.resolveStateMachine:

```
	protected StateMachineModel<S, E> resolveStateMachineModel(String machineId) {
		if (stateMachineModelFactory == null) {
			return defaultStateMachineModel;
		} else {
			StateMachineModel<S, E> m = stateMachineModelFactory.build(machineId);
			if (m.getConfigurationData() == null) {
				// if model doesn't have explicit configuration data,
				// get it from default model
				return new DefaultStateMachineModel<>(defaultStateMachineModel.getConfigurationData(), m.getStatesData(),
						m.getTransitionsData());
			} else {
				return m;
			}
		}
	}
```
This code checks getConfigurationData() returned from the factory build(), and if it is null adds in the configuration from the model (which contains the configuration we want), else the returned configuration is used.  

However the implementation of RepositoryStateMachineModelFactory returns a default (non null) instance of ConfigurationData from getConfigurationData(), thereby preventing the model's configuration from being used.

I believe the problem could be addressed very easily by modifying RepositoryStateMachineModelFactory.build() to return a StateMachineModel with null configuration.

**Comment by jvalkeal** [comment]
Ah yeah, thanks for pointing this out.

**Comment by jvalkeal** [comment]
Oh, looks like this issue is already fixed in `1.2.x` branch with commit https://github.com/spring-projects/spring-statemachine/commit/33a5370d014c05dacc50f1908dc953aa4e65cfb4. Didn't realise this as changes between `1.2.7` and `1.2.8` will be backported to master when `1.2.8` is released.

I'll add some direct tests before closing this issue.

**Comment by jvalkeal** [comment]
.
